### PR TITLE
Data Source for google_compute_security_policy

### PR DESCRIPTION
**Pull request by AarshDhokai**
Fixes the issue : [add google_compute_security_policy data source](https://github.com/hashicorp/terraform-provider-google/issues/3321)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
google_compute_security_policy
```

### CODE REVIEW

**Comment by mitj04** [comment]
LGTM

File: mmv1/third_party/terraform/website/docs/d/compute_security_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+subcategory: "Compute Engine"
+page_title: "Google: google_compute_security_policy"
+description: |-
+  Get information about a Google Compute Security Policy.
+---
+
+# google\_compute\_security\_policy
+
+To get more information about Snapshot, see:
```
Comment (melinath) [inline mmv1/third_party/terraform/website/docs/d/compute_security_policy.html.markdown:10]:
```suggestion
To get more information about Google Compute Security Policy, see:
```

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_security_policy.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	dsSchema := datasourceSchemaFromResourceSchema(resourceComputeSecurityPolicy().Schema)
+	addRequiredFieldsToSchema(dsSchema, "name", "project")
+
+	return &schema.Resource{
+		Read:   dataSourceComputeecurityPolicyRead,
+		Schema: dsSchema,
+	}
+}
+
+func dataSourceComputeecurityPolicyRead(d *schema.ResourceData, meta interface{}) error {
```
Comment (melinath) [inline mmv1/third_party/terraform/data_sources/data_source_google_compute_security_policy.go:19]:
```suggestion
func dataSourceComputeSecurityPolicyRead(d *schema.ResourceData, meta interface{}) error {
```

File: mmv1/third_party/terraform/data_sources/data_source_google_compute_security_policy.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceGoogleComputeSecurityPolicy() *schema.Resource {
+	dsSchema := datasourceSchemaFromResourceSchema(resourceComputeSecurityPolicy().Schema)
+	addRequiredFieldsToSchema(dsSchema, "name", "project")
+
+	return &schema.Resource{
+		Read:   dataSourceComputeecurityPolicyRead,
```
Comment (melinath) [inline mmv1/third_party/terraform/data_sources/data_source_google_compute_security_policy.go:14]:
```suggestion
		Read:   dataSourceComputeSecurityPolicyRead,
```

**Review by melinath — CHANGES_REQUESTED** [review]
In general looks good, just a couple changes to request.

**Comment by melinath** [comment]
closing stale PR

**Pull request closed without merging**